engine/include/Engine/Intermediate/RenderSystem.hpp

```diff
@@ -65,21 +61,39 @@ class RenderSystem
     RenderSystem& operator=(RenderSystem const& other) noexcept = delete;
     RenderSystem& operator=(RenderSystem&& other) noexcept = delete;
 
-    void draw(const ResourceManagerType& res) noexcept;
+public:
+    void tryToBindShader(Shader& shader);
```

Comment:
I like "tryX" functions to return a boolean or to tell  which exception  is throwing to know when the function is not working.